### ISSUE TITLE
n-char mode with timer

### ISSUE THREAD

**Issue by hanschen** [phaazon/hop.nvim#95]
After using [`avy`](https://github.com/abo-abo/avy) in emacs, I came to really like its [`avy-goto-char-timer`](https://github.com/abo-abo/avy#avy-goto-char-timer) mode, which lets you input an arbitrary amount of character within a predefined time, and then lables all matches. If there is a unique match, it will jump directly to the match and disable typing for a short time to avoid accidental inputs.

Here is an example using the [`vim-smalls`](https://github.com/t9md/vim-smalls) plugin (what I currently use):

![timer-mode](https://user-images.githubusercontent.com/1353015/122588017-dc82e080-d05e-11eb-8d09-9491c13c8297.gif)

And this is what I typed to get this result (my trigger key is `-`):
1. `-char<delay>a` (note that `<Return>` is not required)
2. `-famous` (the `mous` part is ignored because `fa` gives a unique match)
3. `-trick<delay>a`

I like this mode because I can be however specific I want, and it feels like one movement because I don't have to press Return. Often it jumps directly to where I want without me having to choose a label.

Here's an example of how to configure `smalls` to achieve this:

```vim
let g:smalls_auto_jump = 1
let g:smalls_auto_jump_min_input_length = 2
let g:smalls_auto_jump_timeout = 0.5  " I personally have it set to 0.2
let g:smalls_auto_set = 1
let g:smalls_auto_set_min_input_length = 1
let g:smalls_exit_at_notfound = 1

nmap - <Plug>(smalls)
```

Related issue: https://github.com/phaazon/hop.nvim/issues/52

**Comment by phaazon** [comment]
This is the first time I see that kind of feature. I don’t plan to use it (as I want Hop to be as reactive as possible). However, it seems interesting, so I have some questions.

Is the _predefined time_ you are mentioning a time out after the last input character? Auto-jumping is agnostic of the method used to jump (it works for any hinting mode). Also, is the timeout “not taken into account” if there is no match?

I don’t really see myself using this, but it looks pretty cool.

**Comment by hanschen** [comment]
> Is the *predefined time* you are mentioning a time out after the last input character? 

Yes that's correct. Something like 0.5 s would be a good default value I think.

> Also, is the timeout “not taken into account” if there is no match?

Yes, that's how I prefer it to work. If there is no match, it will directly exit the mode, flash the word under the cursor (to indicate that it exited), and block user input for a short time to avoid accidental inputs (probably the same amount of time as the *predefined time* above would be good).
